### ISSUE TITLE
Output of visualiser displaying a "0" instead of class name

### ISSUE THREAD

**Issue by dinisov** [facebookresearch/detectron2#2982]
Hi,

In the latest version (0.4) Visualizer.draw_instance_predictions() shows a 0 next to the confidence value, rather than the confidence value alone as in 0.3

I tried to set some metadata for the (non-existing) test dataset with

```
cfg.DATASETS.TEST = ("test",)
MetadataCatalog.get("test").thing_classes = ["my_class"]
```

As per some other replies, but it is still displaying a "0" next to the confidence value. This also happens if I add no metadata, which is not the expected behaviour: if there is only one class only the confidence value should be displayed.

What am I missing here? thanks!

**Comment by dinisov** [comment]
Hmmm... More info? Ok I set up the predictor with:

```
   weights_dir = "weights_dir"

    cfg = get_cfg()
    cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml"))

    cfg.MODEL.WEIGHTS = os.path.join(weights_dir, "model_file.pth")
    cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.7 
    cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1 
    cfg.MODEL.DEVICE='cpu'
    cfg.DATASETS.TEST = ("test",)

    MetadataCatalog.get("test").thing_classes = ["my_class",]
  
    predictor = DefaultPredictor(cfg)
```
And then if I run a prediction on an image with:

`output = predictor(im)`

output is

```
{'instances': Instances(num_instances=2, image_height=979, image_width=986, fields=[pred_boxes: Boxes(tensor([[ 55.7547, 379.2029, 971.4525, 979.0000],
        [459.2875, 265.6748, 479.6541, 289.5044]])), scores: tensor([0.8837, 0.7565]), pred_classes: tensor([0, 0]), pred_masks: tensor([[[False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         ...,
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False]],

        [[False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         ...,
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False],
         [False, False, False,  ..., False, False, False]]])])}
```

Then when I run

`out = v.draw_instance_predictions(output["instances"].to("cpu"))`

The resulting image shows a "0" next to the confidence value, instead of "my_class". This also happens if I comment out the lines setting the metadata. No matter what I do the "0" is there. This exact same code produced the expected result (just the confidence value if there is only one class) in Detectron2 0.3

detectron2 was installed using

`python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'`

And I am running this on a Ubuntu 18.04 machine. The model was trained on two GPUs.

**Comment by ppwwyyxx** [comment]
Please provide complete code as the issue template asks. Variables like `v` is undefined in the given code above.

**Comment by dinisov** [comment]
Hi I updated the code in my comments above as requested. But maybe that is too much information and I made it too complicated.

The function 

`Visualizer.draw_instance_predictions() `

puts a 0 next to the confidence value in 04. That is all. It's probably got nothing to do with my code since the exact same code will not have the "0" in 0.3.

<img width="176" alt="Screen Shot 2021-05-07 at 10 52 06 am" src="https://user-images.githubusercontent.com/9738814/117382619-47021600-af22-11eb-854d-e2503f10cf5e.png">
